<launch>
  <arg name="voxblox_path" default="" />
  <arg name="output_path" default="" />
  <arg name="sparse_graph_path" default="" />
  <arg name="frame_id" default="map" />

  <node name="voxblox_skeleton_server" pkg="voxblox_skeleton" type="skeleton_server_node" output="screen" clear_params="true" respawn="true" args="-v=1">
    <param name="bim_filename" value="/home/paul/spot_ws/src/s_graphs_utils/mav_voxblox_planning/voxblox_skeleton/bims/topfloor_walls_org.json" />

    <!-- global map -->
    <!-- doc: https://voxblox.readthedocs.io/en/latest/pages/The-Voxblox-Node.html -->
    <!-- <param name="max_block_distance_from_body" value="50.0" /> -->

    <!-- <param name="input_filepath" value="$(arg voxblox_path)" />-->
    <!-- <param name="output_filepath" value="$(arg output_path)" />-->
    <!-- <param name="sparse_graph_filepath" value="$(arg sparse_graph_path)" />-->
   
    <!-- voxblox server params -->
    <param name="update_esdf_every_n_sec" value="3.0" />
    <param name="update_mesh_every_n_sec" value="6.0" />
    <param name="color_mode" value="lambert" />
    <param name="traversability_radius" value="0.3" />
    <param name="publish_pointclouds" value="true" />
    <param name="world_frame" value="$(arg frame_id)" />
    <param name="frame_id" value="$(arg frame_id)" />
    <param name="slice_level" value="0.5" />
    <param name="verbose" value="true" />

    <!-- TSDF integrator params -->
    <param name="method" value="fast" />
    <param name="tsdf_voxel_size" value="0.2" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="truncation_distance" value="0.4" />
    <param name="max_ray_length_m" value="10.0" />
    <param name="min_ray_length_m" value="0.1" />
    <param name="max_weight" value="10000" />
    <param name="use_const_weight" value="true" />
    <param name="allow_clear" value="true" />
    <param name="use_freespace_pointcloud" value="false" />

    <!-- TSDF Fast integrator params -->
    <param name="start_voxel_subsampling_factor" value="2" />
    <param name="max_consecutive_ray_collisions" value="2" />
    <param name="max_integration_time_s" value="5" />
    <param name="clear_checks_every_n_frames" value="1" />

    <!-- ESDF integrator params -->
    <param name="esdf_max_distance_m" value="2.0" />
    <param name="esdf_default_distance_m" value="2.0" />
    <param name="clear_sphere_for_planning" value="false" />
    <param name="clear_sphere_radius" value="1.5" />
    <param name="occupied_sphere_radius" value="5.0" />
    <param name="esdf_add_occupied_crust" value="true" />

    <!-- ICP params -->
    <param name="enable_icp" value="false" />

    <!-- skeleton server -->
    <param name="skeleton_publish_data" value="true" />
    <param name="skeleton_visualize_data" value="true" />
    <param name="skeleton_generation_interval" value="3.0" />
    <param name="skeleton_min_separation_angle" value="0.78" />
    <param name="skeleton_generate_by_layer_neighbors" value="false" />
    <param name="skeleton_num_neighbors_for_edge" value="16" />
    <param name="skeleton_min_gvd_distance" value="0.05" />

    <!-- Publish ESDF/TSDF maps for the planner -->
    <param name="publish_esdf_map" value="true" />
    <param name="publish_tsdf_map" value="true" />
    <param name="publish_slices" value="true" />
    <param name="publish_map_every_n_sec" value="1.0" />
  </node>

</launch>
